### PR TITLE
Time offset

### PR DESCRIPTION
**Pull request by jakeware**
This PR represents the first functioning version of SVIS based on the RealSense D435.

### CODE REVIEW

File: svis/src/svis/camera_packet.h [hunk 1]
```diff
@@ -2,17 +2,14 @@
 
 #pragma once
 
-#include "svis/image.h"
-#include "svis/camera_info.h"
 #include "svis/image_metadata.h"
 
 namespace svis {
 
 class CameraPacket {
```
Comment (psiorx) [inline svis/src/svis/camera_packet.h]: Struct might be more appropriate here.

File: svis/src/svis/camera_synchronizer.cc [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+#include <vector>
+#include <limits>
+#include <cmath>
+
+#include "svis/camera_synchronizer.h"
+
+namespace svis {
+
+CameraSynchronizer::CameraSynchronizer()
+  : max_buffer_size_(50) {
```
Comment (psiorx) [inline svis/src/svis/camera_synchronizer.cc]: Would be nice to be able to pass this buffer size into the constructor.

File: svis/src/svis/camera_synchronizer.cc [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+}
+
+SyncState& CameraSynchronizer::GetSyncState(const std::string& sensor_name) {
+  auto pair = sync_states_.find(sensor_name);
+
+  if (pair != sync_states_.end()) {
+    return pair->second;
+  }
+
+  return sync_states_[sensor_name];
```
Comment (psiorx) [inline svis/src/svis/camera_synchronizer.cc]: What is this returning if the sensor wasn't found in the dictionary above?

File: svis/src/svis/sync_state.h [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+#include <deque>
+
+#include "svis/camera_packet.h"
+
+#pragma once
+
+namespace svis {
+
+class SyncState {
```
Comment (psiorx) [inline svis/src/svis/sync_state.h]: Should also be struct if everything is going to be public.

File: svis/src/svis/camera_packet.h (hunk 1 above)
Comment (jakeware) [reply svis/src/svis/camera_packet.h]: True.  77259f17f376c0d5ee272e170021b068998dcc18

File: svis/src/svis/sync_state.h (hunk 4 above)
Comment (jakeware) [reply svis/src/svis/sync_state.h]: https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes

**Pull request merged**